### ISSUE TITLE
Cannot kill a container on Windows Server 2019 running Docker 19.03.1

### ISSUE THREAD

**Issue by mbergal** [moby/moby#39742]
This is what I get in Application Event Log
```
debug: Sending kill signal 9 to container d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c
debug: container kill failed because of 'container not found' or 'no such process' [error=no such container container=d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c action=kill]
```

but `docker inspect` returns

```
[
    {
        "Id": "d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c",
        "Created": "2019-08-06T09:11:04.7342164Z",
        "Path": "cmd /S /C cmd /c ping -t localhost",
        "Args": [],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 9980,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2019-08-06T09:11:08.345674Z",
            "FinishedAt": "0001-01-01T00:00:00Z"
        },
        "Image": "sha256:99db41e70c8ba3064fa18bb92e6312008ae677d094a5fe52348502d13345568c",
        "ResolvConfPath": "",
        "HostnamePath": "",
        "HostsPath": "",
        "LogPath": "C:\\ProgramData\\docker\\containers\\d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c\\d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c-json.log",
        "Name": "/k8s_POD_submission-check-5c4f587ddf-s9lmr_submission-check_003a6b97-e6e3-4af5-9ef7-86a5e63ca817_0",
        "RestartCount": 0,
        "Driver": "windowsfilter",
        "Platform": "windows",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "none",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "shareable",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": -998,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "process",
            "CpuShares": 2,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": null,
            "ReadonlyPaths": null
        },
        "GraphDriver": {
            "Data": {
                "dir": "C:\\ProgramData\\docker\\windowsfilter\\d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c"
            },
            "Name": "windowsfilter"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "submission-check-5c4f587ddf-s9lmr",
            "Domainname": "",
            "User": "ContainerUser",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": [
                "cmd /S /C cmd /c ping -t localhost"
            ],
            "ArgsEscaped": true,
            "Image": "kubeletwin/pause",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "annotation.kubernetes.io/config.seen": "2019-08-06T02:11:04.1232578-07:00",
                "annotation.kubernetes.io/config.source": "api",
                "app": "submission-check",
                "io.kubernetes.container.name": "POD",
                "io.kubernetes.docker.type": "podsandbox",
                "io.kubernetes.pod.name": "submission-check-5c4f587ddf-s9lmr",
                "io.kubernetes.pod.namespace": "submission-check",
                "io.kubernetes.pod.uid": "003a6b97-e6e3-4af5-9ef7-86a5e63ca817",
                "pod-template-hash": "5c4f587ddf"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "d94306d2d22a2b45d9f581f99bae6ac8b4fe249f95e4cfb7bf68890ab506fc2c",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "none": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "8d5951599b7811fb09070ca3e673dba40cd947d0509d02018b6a43d2ce970036",
                    "EndpointID": "e9ae0c0c119451bd447cf7ad3a556cca822d4039b935b7798f6d47301669c415",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]

```

```
Client: Docker Engine - Enterprise
 Version:           19.03.1
 API version:       1.40
 Go version:        go1.12.5
 Git commit:        f660560464
 Built:             07/25/2019 20:59:52
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Enterprise
 Engine:
  Version:          19.03.1
  API version:      1.40 (minimum version 1.24)
  Go version:       go1.12.5
  Git commit:       f660560464
  Built:            07/25/2019 20:57:41
  OS/Arch:          windows/amd64
  Experimental:     false
```

**Comment by drnybble** [comment]
Seeing the same thing -- may occur after rebooting a node in the Swarm.

On one node I have "phantom" containers that are not really running. Note: these containers are doing 'Start-Sleep 3600' so are not up for 2 hours plus I just rebooted this machine...

```
PS C:\Users\Administrator> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
c48989a4a0d9        windns:cache        "powershell -Command…"   2 hours ago         Up 2 hours                              windns_s1.1.xbw43d41kfdkypnr38qi31xnd
e92037504dea        windns:cache        "powershell -Command…"   2 hours ago         Up 2 hours                              windns_s2.1.8v5b0qnssnccc6d3x4pslm3md
P
```

Doing 'docker stop' will hang if you try to stop them.

List the services, you see too many replicas:

```
PS C:\Users\Administrator> docker service ls
ID                  NAME                    MODE                REPLICAS            IMAGE               PORTS
2nkatatcrou3        windns_centralcache     replicated          1/1                 windns:cache
t7miww6mpi87        windns_centralnocache   replicated          1/1                 windns:nocache
ut9udzwd2j5j        windns_s1               replicated          2/1                 windns:cache
kotxpstb59wz        windns_s2               replicated          2/1                 windns:cache
869xyya5lmpo        windns_s3               replicated          1/1                 windns:cache
```

You can get rid of them by stopping Docker, deleting the folders in the "C:\ProgramData\docker\containers" folder and then starting up again.

**Comment by thaJeztah** [comment]
ping @ddebroy @vikramhh PTAL

**Comment by ddebrecenijr** [comment]
Was able to force it to kill by navigating to the containers folder in ProgramData and deleting container folders followed by a docker restart.

**Comment by mkoziel2000** [comment]
I am experiencing the exact same issue with same symptoms and workaround.  I did notice that when creating/starting containers using docker-compose, the rebooting of the docker service or machine does not result in docker thinking the containers are still running.  However, I still need to manually restart them after the reboot even when the restart behavior on the container is set to "always".

**Comment by mbergal** [comment]
ping @ddebroy @vikramhh PTAL

**Comment by StefanScherer** [comment]
I can reproduce the problem with 19.03.1 (docker run -d --restart always ...) and after a reboot the container is not started.
It is fixed with 19.03.2 with PR #39703

**Comment by thaJeztah** [comment]
Thanks @StefanScherer; let me close this issue
